File: SoftUni_JAVA_BASIC_Jan_2023_SolutionsOfProblems/P06Ex_Nested_Loops/P03_SumPrimeNonPrime.java
package P06Ex_Nested_Loops;

import java.util.Scanner;

public class P03_SumPrimeNonPrime {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        int sumPrime = 0, sumNonPrime = 0;
        while (!input.equals("stop")) {
            int number = Integer.parseInt(input);
            boolean divFlag = false;
            if (number < 0) {
                System.out.println("Number is negative.");
            } else {
                for (int divisor = 2; divisor <= number / 2; divisor++) {
                    if (number % divisor == 0) {
                        divFlag = true;
                        break;
                    }
                }
                if (divFlag) {
                    sumNonPrime += number;
                } else {
                    sumPrime += number;
                }
            }
            input = scanner.nextLine();
        }
        System.out.printf("Sum of all prime numbers is: %d%n", sumPrime);
        System.out.printf("Sum of all non prime numbers is: %d%n",sumNonPrime);
    }
}

File: JavaScript_Basic_04_2023_SolutionsOfProblems/06.Ex_ PB-JS-Nested-Loops/P03.SumPrimeNonPrime.js
function sumPrimeNonPrime(input) {
    let sumPrimeNumber = 0;
    let sumNonPrimeNumber = 0;
    let index = 0;

    while (input[index] !== "stop") {
        let currentNumber = Number(input[index]);
        if (currentNumber < 0) {
            console.log("Number is negative.");
            index++;
            continue;
        }
        let isPrime = true;

        for (divisor = 2; divisor <= parseInt(currentNumber / 2); divisor++) {
            if (currentNumber % divisor === 0) {
                isPrime = false;
                break;
            }
        }

        if (isPrime) {
            sumPrimeNumber += currentNumber;
        } else {
            sumNonPrimeNumber += currentNumber;
        }

        index++;
    }
    console.log(`Sum of all prime numbers is: ${sumPrimeNumber}`);
    console.log(`Sum of all non prime numbers is: ${sumNonPrimeNumber}`);
}

sumPrimeNonPrime(["0",
    "-9",
    "0",
    "stop"])


